File: fake-store-quarkus/src/main/java/org/arielzarate/interfaces/rest/mapper/ProductMapper.java
package org.arielzarate.interfaces.rest.mapper;

import org.arielzarate.domain.model.Product;
import org.arielzarate.interfaces.rest.dto.ProductRequest;
import org.arielzarate.interfaces.rest.dto.ProductResponse;

public class ProductMapper {

    public static Product toDomain(ProductRequest request) {
        if (request == null) return null;

        Product product = new Product();
        product.setTitle(request.getTitle());
        product.setPrice(request.getPrice());
        product.setDescription(request.getDescription());
        product.setCategory(request.getCategory());
        product.setImage(request.getImage());

        if (request.getRating() != null) {
            Product.Rating rating = new Product.Rating(
                request.getRating().getRate(),
                request.getRating().getCount()
            );
            product.setRating(rating);
        }
        return product;
    }

    public static ProductResponse toResponse(Product product) {
        if (product == null) return null;

        ProductResponse response = new ProductResponse();
        response.setId(product.getId());
        response.setTitle(product.getTitle());
        response.setPrice(product.getPrice());
        response.setDescription(product.getDescription());
        response.setCategory(product.getCategory());
        response.setImage(product.getImage());

        if (product.getRating() != null) {
            ProductResponse.RatingResponse rating = new ProductResponse.RatingResponse(
                product.getRating().getRate(),
                product.getRating().getCount()
            );
            response.setRating(rating);
        }
        return response;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/fake_api_hexagonal/interfaces/mappers/ProductMapper.kt
package com.arielZarate.fake_api_hexagonal.interfaces.mappers

import com.arielZarate.fake_api_hexagonal.domain.model.Product
import com.arielZarate.fake_api_hexagonal.interfaces.model.ProductRequest
import com.arielZarate.fake_api_hexagonal.interfaces.model.ProductResponse
import org.springframework.stereotype.Component


@Component
class ProductMapper {
    fun mapToDomain(productRequest: ProductRequest): Product {

        return Product(
            title = productRequest.title,
            price = productRequest.price,
            description = productRequest.description,
            category = productRequest.category,
            image = productRequest.image,
            rating = Product.Rating(
                rate = productRequest.rating.rate, count = productRequest.rating.count
            )
        )
    }


    fun mapToResponse(product: Product): ProductResponse {
        return ProductResponse(
            id = product.id ?: 0,
            title = product.title,
            price = product.price,
            description = product.description,
            category = product.category,
            image = product.image,
            rating = ProductResponse.RatingResponse(
                rate = product.rating.rate, count = product.rating.count
            ),
        )
    }
}